spec/models/person_spec.rb

```diff
@@ -38,6 +38,43 @@
     end
   end
 
+  describe ".create_or_update" do
+    context "new record" do
+      it "creates record with appropriate values" do
+        hash = { email:      'user@example.com',
+                 phone:      '555-555-1111',
+                 first_name: 'Bob',
+                 last_name:  'Garfield' }
+        person = Person.create_or_update(hash)
+        expect(person.slice(*hash.keys).values).to eq hash.values
+      end
+    end
+    context "existing record" do
+      it "updates record with appropriate values" do
+        FactoryGirl.create(:person, email: 'user@example.com')
+        hash = { email: 'user@example.com',
+                 phone: '555-555-1111' }
+        person = Person.create_or_update(hash)
+        expect(person.slice(*hash.keys).values).to eq hash.values
+      end
+    end
+  end
+
+  describe "#save_location" do
+
+    it "calls update location if zip present" do
+      expect_any_instance_of(Location).to receive(:update_location).
+        with(address: '2020 Oregon St', zip: '94703') { true }
+      Person.create(email: 'user@example.com', address: '2020 Oregon St', zip: '94703')
```

Comment:
Line is too long. [87/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.